src/components/course/data/hooks.js

```diff
@@ -0,0 +1,185 @@
+        const newSubject = {
+          ...course.subjects[0],
+          url: `${process.env.MARKETING_SITE_URL}/course/subject/${course.subjects[0].slug}`,
+        };
+        setPrimarySubject(newSubject);
+      }
+    }
+  }, [course]);
+
+  return [subjects, primarySubject];
+}
+
+export function useCoursePartners(course) {
+  const [partners, setPartners] = useState([]);
+  const [label, setLabel] = useState(undefined);
+
+  useEffect(() => {
+    if (course && course.owners) {
+      const newOwners = course.owners.map(owner => ({
+        ...owner,
+        fullUrl: `${process.env.MARKETING_SITE_URL}/${owner.marketingUrl}`,
+      }));
+      setPartners(newOwners);
+      if (newOwners.length > 1) {
+        setLabel('Institutions');
+      } else {
+        setLabel('Institution');
+      }
+    }
+  }, [course]);
+
+  return [partners, label];
+}
+
+export function useCourseRunWeeksToComplete(courseRun) {
+  const [weeksToComplete, setWeeksToComplete] = useState(undefined);
+  const [label, setLabel] = useState(undefined);
+
+  useEffect(() => {
+    if (courseRun && courseRun.weeksToComplete) {
+      setWeeksToComplete(courseRun.weeksToComplete);
+      if (courseRun.weeksToComplete > 1 || courseRun.weeksToComplete === 0) {
+        setLabel('weeks');
+      } else {
+        setLabel('week');
+      }
+    }
+  }, [courseRun]);
+
+  return [weeksToComplete, label];
+}
+
+export function useCourseTranscriptLanguages(courseRun) {
+  const [languages, setLanguages] = useState([]);
+  const [label, setLabel] = useState(undefined);
+
+  useEffect(() => {
+    if (courseRun && courseRun.transcriptLanguages) {
+      setLanguages(courseRun.transcriptLanguages);
+      if (courseRun.transcriptLanguages.length > 1) {
+        setLabel('Video Transcripts');
+      } else {
+        setLabel('Video Transcript');
+      }
+    }
+  }, [courseRun]);
+
+  return [languages, label];
+}
+
+export function useUserEnrollments() {
+  const [userEnrollments, setUserEnrollments] = useState([]);
+  useEffect(() => {
+    fetchUserEnrollments()
+      .then((response) => {
+        const { data: enrollmentsData } = response;
+        const transformedEnrollmentsData = camelCaseObject(enrollmentsData);
+        setUserEnrollments(transformedEnrollmentsData);
+      })
+      .catch((error) => {
+        logError(new Error(error));
+      });
+  }, []);
+
+  return [userEnrollments];
+}
+
+export function useUserEntitlements() {
+  const [userEntitlements, setUserEntitlements] = useState([]);
+  useEffect(() => {
+    fetchUserEntitlements()
+      .then((response) => {
+        const { data: entitlementsData } = response;
+        const transformedEntitlementsData = camelCaseObject(entitlementsData.results);
+        setUserEntitlements(transformedEntitlementsData);
+      })
+      .catch((error) => {
+        logError(new Error(error));
+      });
+  }, []);
+
+  return [userEntitlements];
+}
+
+export function useCourseInEnterpriseCatalog({
```

Comment:
This file is also pretty tough in terms of understanding how each different use/set fits in within the page itself, but maybe I'm just not familiar enough with hooks

Reply:
I'm not sure I'm following what you mean by how each different use/set fits in within the page. Can you clarify?

Reply:
Also can't remember what I meant by this... I'm going to assume it was just Friday afternoon struggles so ignore me